lib/build-components/build_components/templates/python/macros.j2

```diff
@@ -0,0 +1,180 @@
+{# [Builder image]
+    Download and install poetry under the root user.
+#}
+{% macro install_poetry(version) %}
+# Install poetry
+# Note: We prefer to install python packages as --user so that the artifacts are easier
+#       to copy out of the builder image into the final image. poetry already installs
+#       itself under the user directory, and we eventually copy the poetry tool and the
+#       installed python packages under the image USER in the resulting image.
+
+# Ensure we have curl so we can install poetry
+RUN apt-get update \
+ && apt-get -y --no-install-recommends install \
+    curl
+
+# Configure pip, mainly to ensure that packages are installed in the user directory.
+COPY pip.conf /etc/pip.conf
+
+# Install poetry under the root user's home directory.
+# On some images "sh" is aliased to "dash" which does not support "set -o pipefail".
+# We use the "exec" form of RUN to delegate this command to bash instead.
+# This is all because we have a pipe in this command and we wish to fail the build
+# if the any command in the pipeline fails.
+RUN ["/bin/bash", "-c", "set -o pipefail && pip install --upgrade pip && curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | POETRY_VERSION={{version}} python && chmod a+x /root/.poetry/bin/poetry"]
+COPY poetry.toml /root/.config/pypoetry/config.toml
+ENV PATH=/root/.poetry/bin:$PATH
+{% endmacro %}
+
+
+{# [Builder image]
+    Install or copy the component.
+#}
+{% macro build_component(component, build_info) %}
+FROM {{build_info.builder_image}} AS builder-{{component}}
+
+# Install packages required to build native library components
+RUN apt-get update \
+ && apt-get -y --no-install-recommends install \
+    build-essential
+# We intentionally do not clear the lists here, as it's unnecessary for a builder image
+# and we may need to install some component-specific components, too.
+# The image size here isn't as much of a concern, since none of these build tools will
+# end-up in the final built image.
+
+{% if build_info.component_packages(component) %}
+# Install any additional packages required by the component
+RUN apt-get -y --no-install-recommends install \
+    {{" ".join(build_info.component_packages(component))}}
```

Comment:
There's no other escaping we need to worry about here, right?

Reply:
i don't think so? did you have something in mind. The hope was that it would just be a pass-through of the `component.yaml` value. I don't think any packages have spaces in their names, which could cause issues here, i suppose.